### DISCUSSION TITLE
Problems with building?

### DISCUSSION THREAD

**Discussion by Ammako** [d0k3/GodMode9#870] [Q&A] [answered]
Hey, I'm trying to build this with custom changes for myself (chainloading firm1 when installed to firm0 as bootloader, not something that should be in an official build for obvious reasons), but I don't seem to be able to make functional builds these days. Is there anything I could check to figure out what might be causing incompatibilities?

I'm building on Arch, I install firmtool by cloning the repo and using `python setup.py install`, because you can't just use pip install directly on Arch without messing with virtual environments and I don't feel like figuring that out right now. I'm not sure if that could be causing an issue?

Latest devkitARM, Python 3.12.4, the build succeeds just fine and I get a .firm, but booting this firm on console leads to a black screen. firmtool says it depends on pycryptodome but it would make the firm just fine even though I hadn't installed that. I installed it and re-built (make clean --> make firm) just in case missing pycryptodome could have caused firmtool to build the firm wrong or something, but this didn't make a difference.

Has anyone else run into this issue? Clearly it's something on my end, and not the code in the repo, but because I don't get any errors anywhere during the build, I can't really troubleshoot anything.

**Comment by urherenow** [comment]
Not trying to be rude, but github is generally for reporting bugs and collaboration, and not a help forum. This is better left for discord and/or an actual forum such as gbatemp.

**Reply by Ammako** [reply]
You were so eager to be an ass for no reason, that you failed to realize I specifically posted this in the questions and answers section of discussions. Move along now.

**Comment by Wolfvak** [comment, marked as answer]
Are you using the latest-est commit? There's a known issue that's been fixed a few weeks ago regarding the latest dkA toolchain (r64-2 iirc)

**Reply by Ammako** [reply]
Cheers, I'll have to look into that. I was building 2.1.1, that I wasn't getting any build errors threw me off and I didn't consider possible incompatibilities with newer toolchain.

(Side note, I would have found the relevant issue about this, if I had searched for "compile" instead of "build"...)

Though mine does have backlight on when it hangs at boot, while profi's issue says no backlight. But that's probably just because I boot it differently (from firm0, and by booting the .firm through gm9)
